### PR TITLE
quality: parameterize branch names and extract test action

### PR DESCRIPTION
**Pull request by typotter**
🎟️ Fixes FF-3094 towards FF-3085

👯‍♂️ **Related PRs**
- [sdk-test-data#58](https://github.com/Eppo-exp/sdk-test-data/pull/58)
- [java-server-sdk#75](https://github.com/Eppo-exp/java-server-sdk/pull/75)
- [js-client-sdk#96](https://github.com/Eppo-exp/js-client-sdk/pull/96)

### Motivation
Changes are often made to the [sdk-test-data](https://github.com/Eppo-exp/sdk-test-data) repository to capture new behaviours, bugs and edge cases. When these changes are pushed to `main`, the SDKs are cloned locally (locally to the github action running) and their respective tests are run. These tests are set up and run by copies of the SDK test workflows - see [sdk-test-data workflow](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows/test-sdks.yml). There are a number of limitations to this setup:

- Test steps are copied from the SDK’s respective workflows; changes to the SDK test workflows need to be replicated in sdk-test-data and this is not obvious to devs
- The SDK tests are not able to run against in-flight changes to `sdk-test-data`
- When new test-data is committed that breaks an SDK, that breakage is not surfaced in the SDK’s repository until some action triggers its testing workflow (on demand, on pull-request, etc.).

### Description of Changes
_This change_
🚀 - The test job is extracted into a composite action, allowing it to be called from both the SDK's local workflow as well as from workflows in other repositories.

_External to this Change_
[sdk-test-data](https://github.com/Eppo-exp/sdk-test-data/blob/main/.github/workflows) get two testing workflows.
1. ♻️  "Local Testing"- For all pull request changes, the "Local Testing" workflow calls the reusable SDK workflows, test results are recorded only in the sdk-test-data action. This is run using the main SDK branch and the "current" branch of workflow, i.e. the pull request branch. (SDK repo does not see/is not notified of test failures during PR lifecycle, only on push to main)
2. ♻️ 🧑‍💻 "Remote Testing" - On all pushes to the main branch of sdk-test-data, the SDK testing workflows are triggered to run within their respective repositories, alerting all subscribers of failed runs (repo is red/green).

### CODE REVIEW

File: .github/workflows/lint-test-sdk.yml [hunk 1]
```diff
@@ -50,17 +51,6 @@ jobs:
...
-        run: |
-          python -VV
-          python -m pip install --upgrade pip setuptools wheel
-          python -m pip install -r requirements.txt
-          python -m pip install -r requirements-test.txt
-      - name: 'Set up GCP SDK to download test data'
-        uses: 'google-github-actions/setup-gcloud@v0'
-      - name: "Run tests"
-        run: make test
+      - uses: Eppo-exp/python-sdk/.github/actions/action-test@tp/workflows/remote
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:54]: temp branch reference.

File: .github/workflows/lint-test-sdk.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
   pull_request:
     paths:
       - '**/*'
+  workflow_dispatch:
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:7]: allows workflow to be triggered remotely via API

File: .github/workflows/lint-test-sdk.yml [hunk 3]
```diff
@@ -50,17 +51,6 @@ jobs:
   run-tests-with-tox:
     runs-on: ubuntu-latest
     steps:
-      - uses: "actions/checkout@v2"
```
Comment (typotter) [inline .github/workflows/lint-test-sdk.yml:53]: extracted into a composite action. Keeps the main workflow file cleaner in exchange for some more boilerplate and yet another yaml file

**Review by leoromanovsky — APPROVED** [review]

File: .github/workflows/lint-test-sdk.yml (hunk 2 above)
Comment (aarsilv) [reply .github/workflows/lint-test-sdk.yml:7]: I like the rename to be consistent with other SDKs as well 🧹

File: .github/workflows/lint-test-sdk.yml (hunk 1 above)
Comment (aarsilv) [reply .github/workflows/lint-test-sdk.yml:54]: Would a relative path work here and be more concise?

File: .github/actions/action-test/action.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+name: 'Test SDK'
```
Comment (aarsilv) [inline .github/actions/action-test/action.yml:1]: Should we rename this file? `action.yml` feels kind of generic

**Review by aarsilv — APPROVED** [review]

File: .github/workflows/lint-test-sdk.yml (hunk 1 above)
Comment (rasendubi) [inline .github/workflows/lint-test-sdk.yml:54]: @typotter you seem to forgot removing temp branch reference before merging? ^

This broke PR testing: https://github.com/Eppo-exp/python-sdk/actions/runs/10634064395/job/29480558050?pr=68
